File: app/src/main/java/com/musicplayer/SocyMusic/data/base/PlaylistDao.java
package com.musicplayer.SocyMusic.data.base;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.musicplayer.SocyMusic.data.Playlist;
import com.musicplayer.SocyMusic.data.Song;

import java.util.List;

@Dao
public interface PlaylistDao {

    @Query("SELECT * FROM Playlist")
    List<Playlist> getAll();

    @Insert
    void insert(Playlist playlist);

    @Query("SELECT song_id,song_path FROM Song NATURAL JOIN Playlist_Song WHERE Playlist_Song.playlist_id = :playlist_id ORDER BY song_index")
    List<Song> getSongs(String playlist_id);

    @Query("SELECT * FROM Playlist WHERE playlist_id=:playlist_id")
    Playlist get(String playlist_id);

    @Update
    void update(Playlist playlist);

}

File: app/src/main/java/com/musicplayer/OpenMusic/data/base/PlaylistDao.kt
package com.musicplayer.OpenMusic.data.base

import androidx.room.*
import com.musicplayer.OpenMusic.data.Playlist
import com.musicplayer.OpenMusic.data.Song

@Dao
interface PlaylistDao {
    @get:Query("SELECT * FROM Playlist")
    val all: ArrayList<Playlist>

    @Insert
    fun insert(playlist: Playlist)

    @Query("SELECT song_id,song_path,album_id FROM Song NATURAL JOIN Playlist_Song WHERE Playlist_Song.playlist_id = :playlist_id ORDER BY song_index")
    fun getSongs(playlist_id: String): ArrayList<Song>

    @Query("SELECT * FROM Playlist WHERE playlist_id=:playlist_id")
    operator fun get(playlist_id: String): Playlist

    @Update
    fun update(playlist: Playlist)

    @Delete
    fun delete(playlist: Playlist)
}